airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -324,3 +370,35 @@ def read_records(
 
         # Always return an empty generator just in case no records were ever yielded
         yield from []
+
+
+class HttpSubStream(HttpStream, ABC):
+
+    def __init__(self, parent: HttpStream, **kwargs):
+        """
+        :param parent: should be the instance of HttpStream class
+        """
+        super().__init__(**kwargs)
+        self.parent = parent
+
+    def stream_slices(
+        self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        parent_stream_slices = self.parent.stream_slices(
+                sync_mode=sync_mode,
```

Comment:
unfortunately, we can't use the same `sync_mode`, because in the case of incremental we might miss records from the child stream (if updating/creating child record doesn't update cursor_field on parent record)